### PR TITLE
Upload metadata on merge to master with GHA

### PR DESCRIPTION
**Pull request by bnchrch**
## What
Use GHA instead of dagger temporarily

### CODE REVIEW

File: .github/workflows/upload-metadata-files.yml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    paths:
+      - "airbyte-integrations/connectors/**/metadata.yaml"
+
+jobs:
+  upload-metadata-files-to-gcs:
+    name: "Upload changed Metadata Files to Prod Bucket"
+    runs-on: ubuntu-latest
+    env:
+      DOCKER_HUB_PASSWORD: ${{ secrets.DOCKER_HUB_PASSWORD }}
+      DOCKER_HUB_USERNAME: ${{ secrets.DOCKER_HUB_USERNAME }}
```
Comment (pedroslopez) [inline .github/workflows/upload-metadata-files.yml:16]: Can we move this to the step that requires it?

**Review by pedroslopez — APPROVED** [review]

File: .github/workflows/upload-metadata-files.yml (hunk 1 above)
Comment (bnchrch) [reply .github/workflows/upload-metadata-files.yml:16]: Good call out!

**Review by alafanechere — CHANGES_REQUESTED** [review]
I think we should also remove this logic from the "publish on merge" if you want to rely on GHA instead of dagger for this job:
https://github.com/airbytehq/airbyte/blob/master/tools/ci_connector_ops/ci_connector_ops/pipelines/publish.py#L222

File: .github/workflows/upload-metadata-files.yml [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+        uses: tj-actions/changed-files@v35
+        with:
+          files: "airbyte-integrations/connectors/**/metadata.yaml"
+      - name: Setup Python 3.10
+        if: steps.changed-files-specific.outputs.any_changed == 'true'
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.10'
+      - name: Install metadata_service
+        if: steps.changed-files-specific.outputs.any_changed == 'true'
```
Comment (alafanechere) [inline .github/workflows/upload-metadata-files.yml:28]: nit: you could add a previous step to cancel the workflow run `steps.changed-files-specific.outputs.any_changed == 'false'`

**Comment by bnchrch** [comment]
@alafanechere Im thinking we should leave the upload metadata step in publish for two reasons
1. Im hoping the GHA is short lived (under 2 weeks at most)
2. I want to be sure that metadata publish works with dagger publish before we release.

Thoughts?

**Review by alafanechere — APPROVED** [review]
> Thoughts?

I was just concern about race conditions / extra triggers that could be caused by running both pipelines in parallel. But the MD5 sensor should prevent extra triggers.

**Comment by bnchrch** [comment]
@alafanechere yup im hoping the MD5 saves us on the double uploads.

Leaving in the metadata upload in the publish pipeline will also save us from forgetting to add the upload step back when we remove `/publish` (if we dont then this GHA action will fail on a race condition if it doesnt find a docker hub image)

**Pull request merged**